spellbook/models/sudoswap/ethereum/sudoswap_ethereum_events.sql

```diff
@@ -0,0 +1,240 @@
+        )
+}}
+
+--base table CTEs
+WITH 
+    swaps as (
+        SELECT 
+            * 
+        FROM (
+            SELECT 
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , tokenRecipient as call_from
+                , 'Sell' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapNFTsForToken') }}
+            UNION ALL
+            SELECT 
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , nftRecipient as call_from
+                , 'Buy' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapTokenForAnyNFTs') }}
+            UNION ALL
+            SELECT
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , nftRecipient as call_from
+                , 'Buy' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapTokenForSpecificNFTs') }}
+        ) s
+        {% if is_incremental() %}
+        -- this filter will only be applied on an incremental run. We only want to update with new swaps.
+        AND s.call_block_time >= (select max(block_time) from {{ this }})
+        {% endif %}
+    ),
+
+    pairs_created as (
+        SELECT 
+            _nft as nftcontractaddress
+            , _initialNFTIDs as nft_ids
+            , _fee as initialfee
+            , _assetRecipient as asset_recip
+            , output_pair as pair_address
+            , call_block_time as block_time
+        FROM {{ source('sudo_amm_ethereum','LSSVMPairFactory_call_createPairETH') }}
+        WHERE call_success
+    ),
+
+    owner_fee_update as (
+        SELECT 
+            *
+        FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_evt_FeeUpdate') }}
+    ),
+
+    protocol_fee_update as (
+        SELECT
+            *
+        FROM {{ source('sudo_amm_ethereum','LSSVMPairFactory_evt_ProtocolFeeMultiplierUpdate') }}
+    ),
+
+    eth_traces as (
+        SELECT 
+            * 
+        FROM {{ source('ethereum', 'traces') }}
+    ),
+
+    eth_transactions as (
+        SELECT 
+            *
+        FROM {{ source('ethereum','transactions') }}
+    ),
+
+    tokens_ethereum_nft as (
+        SELECT 
+            *
+        FROM {{ ref('tokens_ethereum_nft') }}
+    ),
+
+    nft_ethereum_aggregators as (
+        SELECT 
+            *
+        FROM {{ ref('nft_ethereum_aggregators') }}
+    ),
+
+    prices_usd_eth as (
+        SELECT 
+            *
+        FROM {{ source('prices','usd') }}
```

Comment:
i would either apply the incremental filter here or remove the CTE and join directly downstream, with the incremental filter as part of the join